### ISSUE TITLE
Bug: [no-shadow] False positive inside enum

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#5337]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=MYewdgzgLgBAhgI2ALhmArgWwQUwE4wC8MAjANwCwAUNThpjAIIwDe1M8SRMARIsD2oBfakA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uQAsBDACYcA7uijC+0JpHBgAviDlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const abc: number = 1;

enum A {
  abc = "abc"
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-shadow": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error.

"A.abc" does not shadow "abc".

### Actual Result

Getting error for "A.abc"

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.30.5` |
| `@typescript-eslint/parser` | `5.30.5` |
| `TypeScript` | `4.7.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by Josh-Cena** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/2721#issuecomment-718804636

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/5179, https://github.com/typescript-eslint/typescript-eslint/issues/4340, https://github.com/typescript-eslint/typescript-eslint/issues/4310, https://github.com/typescript-eslint/typescript-eslint/issues/2886

Please use the search next time.

We'll add this to the docs as it's something that is frequently misunderstood.

**Comment by ilyub** [comment]
@bradzacher

I examined the explanation that "Enum members declare variables withim the enum scope".

But I have an objection.
It is uncommon pattern to use variables declared within the enum scopes.
You even have a rule to prevent this pattern [@typescript-eslint/prefer-literal-enum-member](https://typescript-eslint.io/rules/prefer-literal-enum-member/) and this rule is marked as recommended.

So, why not add an option to ignore enums in "no-shadow" rule that can be used in combination with "prefer-literal-enum-member" rule:
```
{
  "@typescript-eslint/no-shadow": ["warn", { ignoreEnums: true }],
  "@typescript-eslint/prefer-literal-enum-member": "warn",
}
```

Is it possible to re-open ticket task and consider it as feature request or can I create a new ticket with feature request?

P.S.: I did not find other tickets because they are all closed and I did not remove default filters "is:issue is:open". I will take this into account next time.
